Mathlib/Analysis/Analytic/Constructions.lean

```diff
@@ -724,6 +724,15 @@ lemma AnalyticOnNhd.pow {f : E → A} {s : Set E} (hf : AnalyticOnNhd 𝕜 f s)
     AnalyticOnNhd 𝕜 (fun x ↦ f x ^ n) s :=
   fun _ m ↦ (hf _ m).pow n
 
+/-- Powers of analytic functions (into a normed `𝕜`-algebra) are analytic. -/
+@[fun_prop]
+lemma AnalyticAt.hpow {f : E → A} {z₀ : E} (hf : AnalyticAt 𝕜 f z₀) (n : ℕ) :
+    AnalyticAt 𝕜 (HPow.hPow f n) z₀ := by apply hf.pow
```

Comment:
Is the `pow` version not enough?

Reply:
@girving Thanks for taking the time to look at my PR. It seems that the `fun_prop` tactic needs the lemma on `hpow`, or else it won't be able to prove analyticity even in the simplest cases. The most elementary example is perhaps this, which fails if `AnalyticAt.hpow` is not available.
```
lemma analyticAt_centeredMonomial (z₀ : 𝕜) (n : ℕ) :
    AnalyticAt 𝕜 ((· - z₀) ^ n) z₀ := by fun_prop
```